azure/iac/sql_servers.rego

```diff
@@ -3,36 +3,51 @@ package rule
 # https://docs.microsoft.com/en-us/azure/templates/microsoft.sql/servers
 
 #
-# Always use Private Endpoint for Azure SQL Database and SQL Managed Instance
-#
 
-default sql_public_access = null
+# PR-AZR-0126-ARM
+
+default sql_public_access_disabled = null
+
+azure_attribute_absence["sql_public_access_disabled"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.sql/servers"
+    not resource.properties.publicNetworkAccess
+}
 
-azure_issue["sql_public_access"] {
+azure_issue["sql_public_access_disabled"] {
     resource := input.resources[_]
     lower(resource.type) == "microsoft.sql/servers"
     lower(resource.properties.publicNetworkAccess) != "disabled"
 }
 
-sql_public_access {
+sql_public_access_disabled {
     lower(input.resources[_].type) == "microsoft.sql/servers"
-    not azure_issue["sql_public_access"]
+    not azure_attribute_absence["sql_public_access_disabled"]
+    not azure_issue["sql_public_access_disabled"]
+}
+
+sql_public_access_disabled = false {
```

Comment:
@farchide @shahinshirvani  Please investigate **if `resource.properties.publicNetworkAccess` does not exist**, does public network access get **disabled by default**?
If so, this rule need to be `true`

Reply:
if resource.properties.publicNetworkAccess does not exist, does public network access get enabled by default

Reply:
@shahinshirvani thanks. then the rule is fine.